File: frontend/src/utils/document-status.ts
import type { DocumentStatus } from '@/types/document'

export const documentStatusLabels: Record<DocumentStatus, string> = {
  DRAFT: 'Draft',
  SUBMITTED: 'Submitted',
  UNDER_REVIEW: 'Under review',
  APPROVED: 'Approved',
  REJECTED: 'Rejected',
  ARCHIVED: 'Archived',
}

File: backend/src/main/java/com/digitalarchive/domain/enums/DocumentStatus.java
package com.digitalarchive.domain.enums;

/**
 * Mirrors the Postgres enum type `document_status` created in
 * V1__init_schema.sql.
 * Valid transitions are enforced in the service layer, not here — see
 * slide 11 (state machine) for the allowed graph:
 * DRAFT -> SUBMITTED -> UNDER_REVIEW -> APPROVED -> ARCHIVED
 * -> REJECTED -> DRAFT
 */
public enum DocumentStatus {
    DRAFT,
    SUBMITTED,
    UNDER_REVIEW,
    APPROVED,
    REJECTED,
    ARCHIVED
}
